scanner/families/vulnerabilities/types/vulnerability.go

```diff
@@ -99,60 +122,77 @@ type VulnerabilityDiff struct {
 	ASCIIDiff   string                 `json:"asciiDiff"`
 }
 
-type MergedVulnerability struct {
-	ID            string              `json:"id"` // Merged vulnerability ID used in DiffInfo - THIS IS NOT THE CVE ID
-	Vulnerability Vulnerability       `json:"vulnerability"`
-	Diffs         []VulnerabilityDiff `json:"diffs"`
-	ScannersInfo  []ScannerInfo       `json:"scanners"`
-}
+func mergeSlicesWithoutDuplicates[T comparable](slice1, slice2 []T) []T {
```

Comment:
consider moving this to `to` package, you can then use it as `to.UniqueSlice(append(sliceA, sliceB))`
```suggestion
// UniqueSlice returns a slice without duplicate elements.
func UniqueSlice[T comparable](items []T) []T {
	var filtered []T

	unique := make(map[T]bool, len(items))
	for _, item := range items {
		if !unique[item] {
			filtered = append(filtered, item)
			unique[item] = true
		}
	}

	return filtered
}
```

Reply:
you can include additional method to allow generics too which should simplify existing  `mergeSlicesWithoutDuplicates` and `mergeCVSSSlicesWithoutDuplicates` in the same way

```go
func UniqueGenericSlice[T any](items []T, getKey func(T) string) []T {
	var filtered []T

	unique := make(map[string]bool, len(items))
	for _, item := range items {
		key := getKey(item)
		if key != "" && !unique[key] {
			filtered = append(filtered, item)
			unique[key] = true
		}
	}

	return filtered
}
``` 

Usage example:
```go
mergedLinks := to.UniqueSlice(append(vulnerability.Links, otherVulnerability.Links...))
mergedCVSSs := to.UniqueGenericSlice(append(vulnerability.CVSS, otherVulnerability.CVSS...), func(cvss CVSS) string {
	b, err := json.Marshal(c)
	if err != nil {
		log.Errorf("failed to marshal cvss. c=%+v: %v", c, err)
		return "" // skip
	}

	return string(b)
})

```

Reply:
@paralta please consider using a better name for this method, this one might be difficult for usage, e.g. `UniqueSliceFromKey`